packages/schema/src/config/vite.ts

```diff
@@ -39,7 +39,8 @@ export default defineUntypedSchema({
         if (val) {
           consola.warn('Directly configuring the `vite.publicDir` option is not supported. Instead, set `dir.public`. You can read more in `https://nuxt.com/docs/api/nuxt-config#public`.')
         }
-        return val ?? resolve((await get('srcDir')), (await get('dir')).public)
+        const [srcDir, dir] = await Promise.all([get('srcDir'), get('dir')])
+        return val ?? resolve(srcDir, dir.public)
```

Comment:
This will always resolve these even if they aren't needed - I think you could amend to:

```suggestion
        return val ?? await Promise.all([get('srcDir'), get('dir')])
.then(([srcDir, dir]) => resolve(srcDir, dir.public))
```

Reply:
I have not tested .then performance, does it add any sort of overhead? I can run some benchmarks real quick